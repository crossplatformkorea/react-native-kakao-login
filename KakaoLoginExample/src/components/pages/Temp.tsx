import {
  RootStackNavigationProps,
  RootStackParamList,
} from '../../components/navigations/RootStackNavigator';

import Button from '../uis/Button';
import React from 'react';
import {RouteProp} from '@react-navigation/core';
import styled from 'styled-components/native';
import {useTheme} from 'dooboo-ui';
import {withScreen} from '../../utils/wrapper';

const Container = styled.View`
  flex: 1;
  align-self: stretch;

  background-color: ${({theme}) => theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

interface Props {
  navigation: RootStackNavigationProps<'Temp'>;
  route: RouteProp<RootStackParamList, 'Temp'>;
}

function Page(props: Props): React.ReactElement {
  const {theme} = useTheme();

  const {
    route: {
      params: {param},
    },
    navigation,
  } = props;

  return (
    <Container>
      <Button
        testID="btn-back"
        onPress={(): void => navigation.goBack()}
        text={param}
        style={{
          backgroundColor: theme.text,
        }}
      />
    </Container>
  );
}

export default withScreen(Page);
