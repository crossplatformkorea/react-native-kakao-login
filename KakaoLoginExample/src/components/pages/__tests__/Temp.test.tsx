import 'react-native';

import React, {ReactElement} from 'react';
import {
  RenderAPI,
  act,
  cleanup,
  fireEvent,
  render,
} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import Temp from '../Temp';
import {ThemeType} from 'dooboo-ui';

let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

describe('[Temp] render', () => {
  props = createTestProps({
    route: {
      params: {
        param: 'GO BACK',
      },
    },
  });

  component = createTestElement(<Temp {...props} />);

  afterEach(cleanup);

  it('renders without crashing', () => {
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it('should render [Dark] theme', () => {
    props = createTestProps({
      route: {
        params: {
          param: 'GO BACK',
        },
      },
    });

    component = createTestElement(<Temp {...props} />, ThemeType.DARK);
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('[Temp] Interaction', () => {
  let renderResult: RenderAPI;

  beforeEach(() => {
    renderResult = render(component);
  });

  afterEach(cleanup);

  it('should simulate [onClick] when button has been clicked', () => {
    const btnInstance = renderResult.getByTestId('btn-back');

    act(() => {
      fireEvent.press(btnInstance);
    });

    expect(props.navigation.goBack).toHaveBeenCalled();
  });
});
