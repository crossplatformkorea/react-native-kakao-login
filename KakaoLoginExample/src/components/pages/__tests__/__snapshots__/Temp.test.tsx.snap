// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      activeOpacity={0.7}
      onPress={[Function]}
      testID="btn-back"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#393D7A",
              "borderColor": "#393D7A",
              "borderRadius": 4,
              "borderWidth": 2,
              "height": 52,
              "justifyContent": "center",
              "width": 320,
            },
            Object {
              "backgroundColor": "#000000",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#D3D8E8",
                "fontSize": 14,
              },
            ]
          }
        >
          GO BACK
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Temp] render should render [Dark] theme 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#232323",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      activeOpacity={0.7}
      onPress={[Function]}
      testID="btn-back"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#8A96DC",
              "borderColor": "#8A96DC",
              "borderRadius": 4,
              "borderWidth": 2,
              "height": 52,
              "justifyContent": "center",
              "width": 320,
            },
            Object {
              "backgroundColor": "#D3D8E8",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 14,
              },
            ]
          }
        >
          GO BACK
        </Text>
      </View>
    </View>
  </View>
</View>
`;
